### PR TITLE
⚡️ Pic 1470: Make offender assessments call async

### PR DESCRIPTION
**Pull request by jhevans**

### CODE REVIEW

File: src/main/java/uk/gov/justice/probation/courtcaseservice/service/OffenderService.java [hunk 1]
```diff
@@ -97,35 +97,42 @@ public ProbationRecord getProbationRecord(String crn, boolean applyDocumentFilte
...
-            assessmentsMono.blockOptional().ifPresent(assessments -> {
-                probationRecord.setAssessment(findMostRecentByStatus(assessments).orElse(null));
-            });
-        } catch (OffenderNotFoundException e) {
-            telemetryService.trackApplicationDegradationEvent("assessment data missing from probation record (CRN '" + crn + "' not found in oasys)", e, crn);
-        } catch (Exception e) {
-            telemetryService.trackApplicationDegradationEvent("call failed to get assessment data for for CRN '" + crn + "'", e, crn);
-        }
+        var tuple5 = communityApiMono.blockOptional().orElseThrow(() -> new OffenderNotFoundException(crn));
+        var probationRecord = buildProbationRecord(
```
Comment (chris-faulkner-mojdt) [inline src/main/java/uk/gov/justice/probation/courtcaseservice/service/OffenderService.java:111]: probationRecord is redundant - just return from buildProbationRecord

File: src/main/java/uk/gov/justice/probation/courtcaseservice/service/OffenderService.java [hunk 2]
```diff
@@ -97,35 +97,42 @@ public ProbationRecord getProbationRecord(String crn, boolean applyDocumentFilte
             .collectSortedList(new ConvictionBySentenceComparator()
                                 .thenComparing(Conviction::getConvictionId));
 
-        // This Mono resolves to a 4 tuple containing the convictions (see above), offender managers and CRN documents
+        // This Mono resolves to a 5 tuple containing the convictions (see above), offender managers, CRN documents and assessments
         // As a check against exclusions only we call getOffender which will return 403 status
-        Mono<Tuple4<List<Conviction>, List<OffenderManager>, GroupedDocuments, CommunityApiOffenderResponse>> communityApiMono = Mono.zip(
+        var communityApiMono = Mono.zip(
```
Comment (chris-faulkner-mojdt) [inline src/main/java/uk/gov/justice/probation/courtcaseservice/service/OffenderService.java:102]: Not really just a community API mono but I can't think of anything better :-)   If we are using `var` here then rename the Mono<List<Conviction>> above

**Review by chris-faulkner-mojdt — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/justice/probation/courtcaseservice/service/OffenderService.java (hunk 2 above)
Comment (jhevans) [reply src/main/java/uk/gov/justice/probation/courtcaseservice/service/OffenderService.java:102]: Good spot on these, I had meant to tidy these up but then never got round to it. I'll push a few changes around general tidying and readability

**Review by chris-faulkner-mojdt — DISMISSED** [review]
LGTM

**Review by chris-faulkner-mojdt — APPROVED** [review]
LGTM

**Pull request merged**